src/cascadia/TerminalApp/TerminalWindow.cpp

```diff
@@ -1210,8 +1214,15 @@ namespace winrt::TerminalApp::implementation
         if (_WindowName != value)
         {
             _WindowName = value;
-            _PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"WindowName" });
-            _PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"WindowNameForDisplay" });
+            // If we get initialized with a window name, this will be called
+            // before XAML is stood up, and constructing a
+            // PropertyChangedEventArgs will throw.
+            try
+            {
+                _PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"WindowName" });
+                _PropertyChangedHandlers(*this, Windows::UI::Xaml::Data::PropertyChangedEventArgs{ L"WindowNameForDisplay" });
+            }
+            CATCH_LOG();

```

Comment:
You could alternatively just check if `_root` exists. That way we avoid constructing them in the first place.

Reply:
That's what I tried first, but alas, this is a method on `WindowProperties`, not `TerminalWindow`. So this actually doesn't know if there's a `_root`

Reply:
Oh I didn't even realize that this file has 2 classes.

<details>
<summary>Also, wat.</summary>
This is really just my midnight rambling, but... The fact that there's an entire object dedicated to holding nothing but a single string is just nuts. _Why can't you just pass the thing?_ Nah, you got to create an <code>AbstractStringFactoryFactory</code> first. And then you can't even do this 1950s tube computer level computer science problem at any point? No, you got to create to <code>AbstractObjectFactoryStringFactory</code> first. Did anyony during the design of this realize that this is some "stop get some help" meme tier stuff? lmao. Come on, WinRT/XAML API, just take the X thing from A and stuff it into B <i>as X</i>. No need for Y. Also, wat.
</details>